libmamba/CMakeLists.txt

```diff
@@ -0,0 +1,320 @@
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/core/url.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/core/util.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/core/util_os.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/core/validate.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/core/virtual_packages.hpp
+    # API (high-level)
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/api/c_api.h
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/api/clean.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/api/config.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/api/configuration.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/api/constants.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/api/create.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/api/info.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/api/install.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/api/list.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/api/remove.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/api/shell.hpp
+    ${LIBMAMBA_INCLUDE_DIR}/libmamba/api/update.hpp
+)
+
+# Targets and link
+# ================
+
+macro(libmamba_create_target target_name linkage output_name)
+    string(TOUPPER "${linkage}" linkage_upper)
+
+    if (NOT ${linkage_upper} MATCHES "^(SHARED|STATIC)$")
+        message(FATAL_ERROR "Invalid library linkage: ${linkage}")
+    endif ()
+
+    # Output
+    # ======
+
+    add_library(${target_name} ${linkage_upper} ${LIBMAMBA_SOURCES} ${LIBMAMBA_HEADERS} ${LIBMAMBA_THIRD_PARTIES})
+    if (NOT STATIC_DEPENDENCIES)
+        find_library(LIBSOLV_LIBRARIES NAMES solv)
+        find_library(LIBSOLVEXT_LIBRARIES NAMES solvext)
+        find_package(CURL REQUIRED)
+        find_package(LibArchive REQUIRED)
+        find_package(OpenSSL REQUIRED)
+        find_package(yaml-cpp CONFIG REQUIRED)
+        find_package(reproc++ CONFIG REQUIRED)
+
+
+        set(LIBMAMBA_LIBRARIES_DEPS
+            ${LIBSOLV_LIBRARIES}
+            ${LIBSOLVEXT_LIBRARIES}
+            ${LibArchive_LIBRARIES}
+            ${CURL_LIBRARIES}
+            ${OPENSSL_LIBRARIES}
+            ${YAML_CPP_LIBRARIES}
+            reproc++
+        )
+
+        target_link_libraries(${target_name} PUBLIC
+                              ${LIBMAMBA_LIBRARIES_DEPS}
+                              ${MAMBA_FORCE_DYNAMIC_LIBS}
+                              nlohmann_json::nlohmann_json
+                              Threads::Threads)
+    endif()
+    set_property(TARGET ${target_name} PROPERTY CXX_STANDARD 17)
+
+    target_include_directories(
+        ${target_name}
+        PUBLIC
+            $<BUILD_INTERFACE:${LIBMAMBA_INCLUDE_DIR}>
+            $<INSTALL_INTERFACE:include>
+    )
+
+    target_include_directories(
+        ${target_name}
+        PRIVATE ${LIBMAMBA_SOURCE_DIR}
+    )
+
+    set_target_properties(
+        ${target_name}
+        PROPERTIES
+        PUBLIC_HEADER "${LIBMAMBA_HEADERS}"
+        COMPILE_DEFINITIONS "LIBMAMBA_EXPORTS"
+        PREFIX ""
+        VERSION ${LIBMAMBA_BINARY_VERSION}
+        SOVERSION ${LIBMAMBA_BINARY_CURRENT}
```

Comment:
The `SOVERSION` and `VERSION` properties should be slightly different. See [xeus](https://github.com/jupyter-xeus/xeus/blob/master/CMakeLists.txt#L332) for an example of how to set them?

Reply:
I implemented that @JohanMabille thanks, do you have any good pointer to understand better how it works?